components/director/internal/domain/runtime/repository_test.go

```diff
@@ -0,0 +1,211 @@
+	//then
+	require.NoError(t, err)
+	assert.NoError(t, sqlMock.ExpectationsWereMet())
+	assert.Equal(t, runtimeID, modelRuntime.ID)
+	assert.Equal(t, tenantID, modelRuntime.Tenant)
+}
+
+func TestPgRepository_List_ShouldReturnRuntimeModelsForRuntimeEntities(t *testing.T) {
+	// given
+	runtime1ID := uuid.New().String()
+	runtime2ID := uuid.New().String()
+	tenantID := uuid.New().String()
+
+	timestamp, err := time.Parse(time.RFC3339, "2002-10-02T10:00:00-05:00")
+	require.NoError(t, err)
+
+	sqlxDB, sqlMock := mockDatabase(t)
+
+	rows := sqlmock.NewRows([]string{"id", "tenant_id", "name", "description", "status_condition", "status_timestamp", "auth"}).
+		AddRow(runtime1ID, tenantID, "Runtime ABC", "Description for runtime ABC", "INITIAL", timestamp, agentAuthStr).
+		AddRow(runtime2ID, tenantID, "Runtime XYZ", "Description for runtime XYZ", "INITIAL", timestamp, agentAuthStr)
+
+	sqlMock.ExpectQuery(`^SELECT (.+) FROM "public"."runtimes" WHERE "tenant_id" = \$1$`).
+		WithArgs(tenantID).
+		WillReturnRows(rows)
+
+	ctx := decorateContextWithDB(context.TODO(), sqlxDB)
+
+	pgRepository := runtime.NewPostgresRepository()
+
+	// when
+	modelRuntimePage, err := pgRepository.List(ctx, tenantID, nil, nil, nil)
+
+	//then
+	assert.NoError(t, err)
+	assert.Equal(t, 2, modelRuntimePage.TotalCount)
+	require.NoError(t, sqlMock.ExpectationsWereMet())
+
+	assert.Equal(t, runtime1ID, modelRuntimePage.Data[0].ID)
+	assert.Equal(t, tenantID, modelRuntimePage.Data[0].Tenant)
+
+	assert.Equal(t, runtime2ID, modelRuntimePage.Data[1].ID)
+	assert.Equal(t, tenantID, modelRuntimePage.Data[1].Tenant)
+}
+
+func TestPgRepository_Create_ShouldCreateRuntimeEntityFromValidModel(t *testing.T) {
+	// given
+	runtimeID := uuid.New().String()
+	tenantID := uuid.New().String()
+	timestamp, err := time.Parse(time.RFC3339, "2002-10-02T10:00:00-05:00")
+	assert.NoError(t, err)
+
+	description := "Description for runtime BCD"
+	modelRuntime := &model.Runtime{
+		ID:          runtimeID,
+		Tenant:      tenantID,
+		Name:        "Runtime XYZ",
+		Description: &description,
+		Status: &model.RuntimeStatus{
+			Condition: model.RuntimeStatusConditionInitial,
+			Timestamp: timestamp,
+		},
+		AgentAuth: &model.Auth{
+			Credential: model.CredentialData{
+				Basic: &model.BasicCredentialData{
+					Username: "foo",
+					Password: "bar",
+				},
+			},
+		},
+	}
+
+	sqlxDB, sqlMock := mockDatabase(t)
+
+	sqlMock.ExpectExec(`^INSERT INTO "public"."runtimes" \(.+\) VALUES \(.+\)$`).
+		WithArgs(modelRuntime.ID, modelRuntime.Tenant, modelRuntime.Name, modelRuntime.Description, modelRuntime.Status.Condition, modelRuntime.Status.Timestamp, agentAuthStr).
+		WillReturnResult(sqlmock.NewResult(-1, 1))
+
+	ctx := decorateContextWithDB(context.TODO(), sqlxDB)
+
+	pgRepository := runtime.NewPostgresRepository()
+
+	// when
+	err = pgRepository.Create(ctx, modelRuntime)
+
+	// then
+	assert.NoError(t, err)
+}
+
+func TestPgRepository_Update_ShouldUpdateRuntimeEntityFromValidModel(t *testing.T) {
```

Comment:
I see that there are many edge cases not covered by the tests, do we want to test them as well? (not only in this function)